pep-0272.txt

```diff
@@ -5,182 +5,194 @@ Last-Modified: $Date$
+PEP.)
+
+The CFB mode operates on segments of the plaintext and ciphertext
+that are 'segment_size' bits long.  Therefore, when using this
+mode, the input and output strings must be a multiple of
+'segment_size' bits in length.  'segment_size' must be an integer
+between 1 and block_size\*8, inclusive.  (The factor of 8 comes
+from 'block_size' being measured in bytes and not in bits).  The
+default value for this parameter should be block_size\*8.
+Implementors are allowed to constrain 'segment_size' to be a
+multiple of 8 for simplicity, but they're encouraged to support
+arbitrary values for generality.
+
+Secret-key encryption modules should define two variables:
+
+- block_size
+
+  An integer value; the size of the blocks encrypted by this
+  module, measured in bytes.  For all feedback modes, the length
+  of strings passed to the encrypt() and decrypt() must be a
+  multiple of the block size.
+
+- key_size
+
+  An integer value; the size of the keys required by this
+  module, measured in bytes.  If key_size is None, then the
+  algorithm accepts variable-length keys.  This may mean the
+  module accepts keys of any random length, or that there are a
+  few different possible lengths, e.g. 16, 24, or 32 bytes.  You
+  cannot pass a key of length 0 (that is, the null string '') as
+  a variable-length key.
+
+
+Cipher objects should have two attributes:
+
+- block_size
+
+  An integer value equal to the size of the blocks encrypted by
+  this object.  For algorithms with a variable block size, this
+  value is equal to the block size selected for this object.
+
+- IV
+
+  Contains the initial value which will be used to start a
+  cipher feedback mode; it will always be a string exactly one
+  block in length.  After encrypting or decrypting a string,
+  this value is updated to reflect the modified feedback text.
+  It is read-only, and cannot be assigned a new value.
+
+
+Cipher objects require the following methods:
+
+- decrypt(string)
+
+  Decrypts 'string', using the key-dependent data in the object
+  and with the appropriate feedback mode.  The string's length
+  must be an exact multiple of the algorithm's block size or, in
+  CFB mode, of the segment size.  Returns a string containing
+  the plaintext.
+
+- encrypt(string)
+
+  Encrypts a non-empty string, using the key-dependent data in
+  the object, and with the appropriate feedback mode.  The
+  string's length must be an exact multiple of the algorithm's
+  block size or, in CFB mode, of the segment size.  Returns a
+  string containing the ciphertext.
+
+Here's an example, using a module named 'DES'::
 
     >>> import DES
     >>> obj = DES.new('abcdefgh', DES.MODE_ECB)
     >>> plaintext = "Guido van Rossum is a space alien."
     >>> len(plaintext)
     34
     >>> obj.encrypt(plaintext)
-    Traceback (innermost last):
-      File "<stdin>", line 1, in ?
+    Traceback (innermost last)::
+        File "<stdin>", line 1, in <module>
```

Comment:
If we want to modernize the traceback, we also need to update `Traceback (innermost last):`. But I think it's better to keep the old format:

```
Traceback (innermost last):
  File "<stdin>", line 1, in ?
ValueError: Strings for DES must be a multiple of 8 in length
```